### DISCUSSION TITLE
Is it possible to override home path ?

### DISCUSSION THREAD

**Discussion by yroffin** [pglombardo/PasswordPusher#555] [Q&A] [answered]
i've test with docker PasswordPusher and it's great
by default app listen on '/' ... is it possible to listen on another path ex: '/pwpush' ...
i can put a nginx in front of pwpush ... but it could be nice to change this by config :)

**Comment by yroffin** [comment]
i've just seen you've done the [304_relative_root](https://github.com/pglombardo/PasswordPusher/compare/304_relative_root)
it's seem to be the solution :) ... you don't want to merge it ?

**Comment by pglombardo** [comment]
Sure - I'll merge & release #306 soon.

**Comment by pglombardo** [comment]
#306 has been merged and released.  [v1.21.0](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.21.0) Docker containers are building now.

Not sure if it will be helpful, but there is also a simple nginx & Password Pusher [example here](https://github.com/pglombardo/PasswordPusher/tree/master/containers/examples/pwpush-and-nginx).

Let me know if you hit any issues!

**Comment by pglombardo** [comment, marked as answer]
Summarising as a designated answer for this discussion.

You can configure the application to run in a subfolder by setting the `relative_root` option in `config/settings.yml` or by setting the environment variable `PWP__RELATIVE_ROOT`.  See the [Configuration document](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#application-wide) for more details.

With a setting of `PWP__RELATIVE_ROOT=pwpush`, the application front page would be available at `/pwpush` instead of the default `/`.

This option is available since [release v1.21.0](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.21.0).

See also the nginx & Password Pusher [example here](https://github.com/pglombardo/PasswordPusher/tree/master/containers/examples/pwpush-and-nginx).

**Reply by yroffin** [reply]
something like that should work ?

docker run -d --rm -ePWP__RELATIVE_ROOT=http://nip-127-0-0-1.nip.io:8080/pwpush -p8080:5100 pglombardo/pwpush-ephemeral:1.21.0

**Reply by yroffin** [reply]
i've put debug

D, [2022-12-22T08:06:28.197689 #32] DEBUG -- : source=rack-timeout id=fd4b2e76-a28a-493f-a8da-2f566d84beb3 timeout=15000ms state=ready
D, [2022-12-22T08:06:28.197917 #32] DEBUG -- : source=rack-timeout id=fd4b2e76-a28a-493f-a8da-2f566d84beb3 timeout=15000ms service=0ms state=active
F, [2022-12-22T08:06:28.200624 #32] FATAL -- : [fd4b2e76-a28a-493f-a8da-2f566d84beb3]
[fd4b2e76-a28a-493f-a8da-2f566d84beb3] ActionController::RoutingError (No route matches [GET] "/pwpush"):
[fd4b2e76-a28a-493f-a8da-2f566d84beb3]
D, [2022-12-22T08:06:28.201228 #32] DEBUG -- : source=rack-timeout id=fd4b2e76-a28a-493f-a8da-2f566d84beb3 timeout=15000ms service=4ms state=completed

**Reply by pglombardo** [reply]
`docker run -d --rm -e PWP__RELATIVE_ROOT=pwpush -p8080:5100 pglombardo/pwpush-ephemeral:1.21.0`

Set the [env var only to the relative root](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#application-wide) (subfolder) - not the fully qualified URL.  👍
